src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { setLocationsTask, SiteOSStep, WebsiteOS } from 'vscode-azureappservice';
+import { AzureWizardPromptStep, IWizardOptions } from 'vscode-azureextensionui';
+import { ext } from '../../../extensionVariables';
+import { localize } from '../../../localize';
+import { IFunctionAppWizardContext } from '../IFunctionAppWizardContext';
+import { getStackPicks } from './getStackPicks';
+
+export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWizardContext> {
+    public async prompt(context: IFunctionAppWizardContext): Promise<void> {
+        const placeHolder: string = localize('selectRuntimeStack', 'Select a runtime stack.');
+        context.newSiteStack = (await ext.ui.showQuickPick(getStackPicks(context), { placeHolder })).data;
+
+        if (!context.newSiteStack.minorVersion.stackSettings.linuxRuntimeSettings) {
+            context.newSiteOS = WebsiteOS.windows;
+        } else if (!context.newSiteStack.minorVersion.stackSettings.windowsRuntimeSettings) {
+            context.newSiteOS = WebsiteOS.linux;
+        } else if (!context.advancedCreation) {
+            context.newSiteOS = <WebsiteOS>context.newSiteStack.stack.preferredOs;
+        }
+    }
+
+    public shouldPrompt(context: IFunctionAppWizardContext): boolean {
+        return !context.newSiteStack;
+    }
+
+    public async getSubWizard(context: IFunctionAppWizardContext): Promise<IWizardOptions<IFunctionAppWizardContext> | undefined> {
+        if (context.newSiteOS === undefined) {
+            return { promptSteps: [new SiteOSStep()] };
+        } else {
+            await setLocationsTask(context);
```

Comment:
This seems like a behavioral change.  Before `setLocationsTask` was always called in SubscriptionTreeItem if  `!advancedCreation` but `newSiteOS` can be set in either scenario, right?

Reply:
It's also set in `SiteOSStep` [here](https://github.com/microsoft/vscode-azuretools/blob/main/appservice/src/createAppService/SiteOSStep.ts#L21). In the past we _always_ prompted for OS during advanced create, but now we don't. Thus I have to `setLocationsTask` here to maintain existing behavior.

Reply:
Err actually that was true for App Service - Functions is a bit different. It'll overwrite the locationsTask after the "HostPlan" step [here](https://github.com/microsoft/vscode-azurefunctions/blob/b6d5cd3a140338de5351fdeefafd33ef65af1213/src/commands/createFunctionApp/FunctionAppHostingPlanStep.ts#L21) (which happens after we prompt for runtime/OS).

So setting it here isn't technically necessary, but it's consistent with App Service and doesn't really hurt anything.